Northeastern-Electric-Racing/FinishLine
# 2942 onboarding create toggle checklist endpoint

## Changes

-added new endpoint /onboarding/checklists/item/:checklistId/checked in onboarding.routes.ts to toggle the usersChecked. 
-added toggleChecklistItem in onboarding controller and onboarding service
-status of a checklist item for a specific user.
-implemented logic to update parent checklist item if all child checklist items are checked/unchecked.
-throws exceptions if checklistId doesn't exist or is invalid.
-wrote tests

## Test Cases

- toggle a checklist item (add the current user to usersChecked) and check database updates.
- remove the user from usersChecked and check database updates.
- toggle a child item and verify that the parent checklist item is automatically updated when all child items are checked.
- test with a fake checklistId and made sure error (NotFoundException) is thrown.
- edge case: when toggled again, shouldn't throw error and parent checklist should uncheck itself.

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x ] All commits are tagged with the ticket number
- [x ] No linting errors / newline at end of file warnings
- [x ] All code follows repository-configured prettier formatting
- [x ] No merge conflicts
- [x ] All checks passing
- [x ] Screenshots of UI changes (see Screenshots section)
- [x ] Remove any non-applicable sections of this template
- [ x] Assign the PR to yourself
- [x ] No `yarn.lock` changes (unless dependencies have changed)
- [x ] Request reviewers & ping on Slack
- [ x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)
#2942
